libs/books/data-access/src/lib/+state/reading-list.reducer.ts

```diff
@@ -47,11 +47,17 @@ const readingListReducer = createReducer(
       error: action.error
     };
   }),
-  on(ReadingListActions.addToReadingList, (state, action) =>
+  on(ReadingListActions.confirmedAddToReadingList, (state, action) =>
     readingListAdapter.addOne({ bookId: action.book.id, ...action.book }, state)
   ),
-  on(ReadingListActions.removeFromReadingList, (state, action) =>
+  on(ReadingListActions.failedAddToReadingList, (state, action) =>
```

Comment:
Please go through data access flow and check if failedAddToReadingList and failedRemoveFromReadingList has correct logic.